src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java

```diff
@@ -621,6 +632,51 @@ public void testLaunchHICase() throws Exception {
     }
   }
 
+  @SneakyThrows
+  private void checkRejectInvalidLaunchCombo(String expectedMsg) {
+    formType = "CE";
+    try {
+      doLaunchTest(UUID_0, "CE", false);
+      fail();
+    } catch (CTPException e) {
+      assertEquals(Fault.BAD_REQUEST, e.getFault());
+      assertTrue(e.getMessage(), e.getMessage().contains(expectedMsg));
+    }
+  }
+
+  @SneakyThrows
+  private void checkRejectCeManagerFormFromUnitRegion() {
+    addressLevel = "U";
+    checkRejectInvalidLaunchCombo(
+        "A CE Manager form can only be launched against an establishment address not a UNIT.");
+  }
+
+  @Test
+  public void shouldRejectCeManagerFormFromUnitRegionEast() {
+    region = Region.E;
+    checkRejectCeManagerFormFromUnitRegion();
+  }
+
+  @Test
+  public void shouldRejectCeManagerFormFromUnitRegionWest() {
+    region = Region.W;
+    checkRejectCeManagerFormFromUnitRegion();
+  }
+
+  @Test
+  public void shouldRejectCeManagerFormFromUnitRegionNorth() {
+    region = Region.N;
```

Comment:
I'm not keen on the use of globals and think it would be better if they are removed. I think that these tests would be a bit clearer if they passed the region that they are testing down into the test method.
So this code would become 'checkRejectCeManagerFormFromUnitRegion(Region.N)'.
I appreciate that the globals reduce the argument lists but they do introduce an element of invisibility between the test method itself and the lower level method which is actually testing some permutation. When you read a lower level method the fields being tweaked for the callers purpose are not readily visible as they are semi hidden in a subset of the globals.
This comment applies to the other globals.

Reply:
I'll see what I can do

Reply:
OK refactored to remove globals. Not convinced it makes the tests easier to understand, but hopefully you feel that way Pete :)